sdk/internal/recording/recording.go

```diff
@@ -412,3 +417,275 @@ var modeMap = map[RecordMode]recorder.Mode{
+	return cwd
+}
+
+func StartRecording(t *testing.T, options *RecordingOptions) error {
+	if options == nil {
+		options = defaultOptions()
+	}
+	if recordMode == "" {
+		t.Log("AZURE_RECORD_MODE was not set, options are \"record\" or \"playback\". \nDefaulting to playback")
+		recordMode = "playback"
+	} else {
+		t.Log("AZURE_RECORD_MODE: ", recordMode)
+	}
+	testId := getTestId(t)
+	// cwd, err := os.Getwd()
+	// fmt.Printf("CWD: %v\n", cwd)
+	// fmt.Printf("Test recording ID: %v\n", testId)
+
+	url := fmt.Sprintf("%v/%v/start", options.HostScheme(), recordMode)
+
+	req, err := http.NewRequest("POST", url, nil)
+	if err != nil {
+		return err
+	}
+
+	req.Header.Set("x-recording-file", testId)
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return err
+	}
+	recordingId = resp.Header.Get(IdHeader)
+	return nil
+}
+
+func StopRecording(t *testing.T, options *RecordingOptions) error {
+	if options == nil {
+		options = defaultOptions()
+	}
+
+	url := fmt.Sprintf("%v/%v/stop", options.HostScheme(), recordMode)
+	req, err := http.NewRequest("POST", url, nil)
+	if err != nil {
+		return err
+	}
+	if recordingId == "" {
+		return errors.New("Recording ID was never set. Did you call StartRecording?")
+	}
+	req.Header.Set("x-recording-id", recordingId)
+	_, err = client.Do(req)
+	if err != nil {
+		t.Errorf(err.Error())
+	}
+	return nil
+}
+
+func AddUriSanitizer(replacement, regex string, options *RecordingOptions) error {
+	if options == nil {
+		options = defaultOptions()
+	}
+	url := fmt.Sprintf("%v/Admin/AddSanitizer", options.HostScheme())
+	req, err := http.NewRequest("POST", url, nil)
+	if err != nil {
+		return err
+	}
+	req.Header.Set("x-abstraction-identifier", "UriRegexSanitizer")
+	bodyContent := map[string]string{
+		"value": replacement,
+		"regex": regex,
+	}
+	marshalled, err := json.Marshal(bodyContent)
+	if err != nil {
+		return err
+	}
+	req.Body = ioutil.NopCloser(bytes.NewReader(marshalled))
+	req.ContentLength = int64(len(marshalled))
+	_, err = client.Do(req)
+	return err
+}
+
+func (o *RecordingOptions) Init() {
+	if o.MaxRetries != 0 {
+		o.MaxRetries = 0
+	}
+	if o.UseHTTPS {
+		o.Host = baseProxyURLSecure
+		o.Scheme = "https"
+	} else {
+		o.Host = baseProxyURL
+		o.Scheme = "http"
+	}
+}
+
+// type recordingPolicy struct {
+// 	options RecordingOptions
+// }
+
+// func NewRecordingPolicy(o *RecordingOptions) azcore.Policy {
+// 	if o == nil {
+// 		o = &RecordingOptions{}
+// 	}
+// 	p := &recordingPolicy{options: *o}
+// 	p.options.init()
+// 	return p
+// }
+
+// func (p *recordingPolicy) Do(req *azcore.Request) (resp *azcore.Response, err error) {
+// 	originalURLHost := req.URL.Host
+// 	req.URL.Scheme = "https"
+// 	req.URL.Host = p.options.host
+// 	req.Host = p.options.host
+
+// 	req.Header.Set(UpstreamUriHeader, fmt.Sprintf("%v://%v", p.options.scheme, originalURLHost))
+// 	req.Header.Set(ModeHeader, recordMode)
+// 	req.Header.Set(recordingIdHeader, recordingId)
+
+// 	return req.Next()
+// }
```

Comment:
I would love to have this commented out section in azcore because it is going to be used by every test suite that runs against the test-proxy, however it creates a circular dependency. @jhendrixMSFT is there another way we could do this without creating a circular dependency between internal and azcore? We can break out functionality used strictly for testing as it's own library (but would have to require azcore to not use this functionality)